src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2324,19 +2321,9 @@ private void env(String name, String value) {
         else                environment.put(name,value);
     }
 
-    /**
-     * Returns the {@link org.eclipse.jgit.lib.Repository} used by this git instance.
-     *
-     * @return a {@link org.eclipse.jgit.lib.Repository} object.
-     * @throws hudson.plugins.git.GitException if underlying git operation fails.
-     */
-    @NonNull
-    public Repository getRepository() throws GitException {
-        try {
-            return FileRepositoryBuilder.create(new File(workspace, Constants.DOT_GIT));
-        } catch (IOException e) {
-            throw new GitException("Failed to open Git repository " + workspace, e);
-        }
+    @Override
+    public <T> T withRepository(final RepositoryCallback<T> callable) throws IOException, InterruptedException {
+        return workspace.act(new WithRepository<>(callable));
```

Comment:
Of course we should be discouraging these anyway—it is pretty confusing to request CLI Git and then see JGit being sent to the agent.